lib/generic/IdleEffect.cpp

```diff
@@ -16,6 +17,8 @@ void IdleEffect::DoStep() {
       3, 0, CHSV(hue + 128, 255, Brightness(angle + 128)));
   poles[Pole::kNumPoles - 2].SetGridLight(
       3, 0, CHSV(hue + 192, 255, Brightness(angle + 192)));
+
+  PoleEffect::SetPoleEffect(poles, millis());
```

Comment:
I would prefer to pass the timestamp into this constructor instead of grabbing it here. Doing that everywhere would ensure that we get a consistent timestamp for each operation, and it also makes it possible to test this constructor with specific timestamps.

Reply:
What do you mean "this constructor"? You mean pass the timestamp into `Effect::DoStep`? I can refactor all the Effects classes to handle it in that case.

Reply:
Ack, I'll do this in a follow up as the rest of the code accesses `millis()` directly and it's outside the scope of this PR.